README.md

```diff
@@ -1,11 +1,432 @@
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Contents**
+
+- [Quick Start](#quick-start)
+  - [Installation](#installation)
+  - [Implementing an Activity](#implementing-an-activity)
+  - [Running a Workflow](#running-a-workflow)
+- [Usage](#usage)
+  - [Client](#client)
+    - [Cloud Client Using mTLS](#cloud-client-using-mtls)
+    - [Data Conversion](#data-conversion)
+      - [ActiveRecord and ActiveModel](#activerecord-and-activemodel)
+  - [Workers](#workers)
+  - [Workflows](#workflows)
+  - [Activities](#activities)
+    - [Activity Definition](#activity-definition)
+    - [Activity Context](#activity-context)
+    - [Activity Heartbeating and Cancellation](#activity-heartbeating-and-cancellation)
+    - [Activity Worker Shutdown](#activity-worker-shutdown)
+    - [Activity Concurrency and Executors](#activity-concurrency-and-executors)
+    - [Activity Testing](#activity-testing)
+- [Development](#development)
+  - [Build](#build)
+  - [Testing](#testing)
+  - [Code Formatting and Type Checking](#code-formatting-and-type-checking)
+  - [Proto Generation](#proto-generation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Quick Start
+
+### Installation
+
+⚠️ PENDING GEM PUBLISH
+
+**NOTE: Due to [an issue](https://github.com/temporalio/sdk-ruby/issues/162), fibers (and `async` gem) are only
+supported on Ruby versions 3.3 and newer.
+
+### Implementing an Activity
+
+Implementing workflows is not yet supported in the Ruby SDK, but implementing activities is.
+
+For example, if you have a `SayHelloWorkflow` workflow in another Temporal language that invokes `SayHello` activity on
+`my-task-queue` in Ruby, you can have the following Ruby script:
+
+```ruby
+require 'temporalio/activity'
+require 'temporalio/cancellation'
+require 'temporalio/client'
+require 'temporalio/worker'
+
+# Implementation of a simple activity
+class SayHelloActivity < Temporalio::Activity
+  def execute(name)
+    "Hello, #{name}!"
+  end
+end
+
+# Create a client
+client = Temporalio::Client.connect('localhost:7233', 'my-namespace')
+
+# Create a worker with the client and activities
+worker = Temporalio::Worker.new(
+  client:,
+  task_queue: 'my-task-queue',
+  # There are various forms an activity can take, see specific section for details.
+  activities: [SayHelloActivity]
+)
+
+# Run the worker until SIGINT. This can be done in many ways, see specific
+# section for details.
+cancellation, cancel_proc = Temporalio::Cancellation.new
+Signal.trap('INT') { cancel_proc }
+worker.run(cancellation:)
+```
+
+Running that will run the worker until Ctrl+C pressed.
+
+### Running a Workflow
+
+Assuming that `SayHelloWorkflow` just calls this activity, it can be run like so:
+
+```ruby
+require 'temporalio/client'
+
+# Create a client
+client = Temporalio::Client.connect('localhost:7233', 'my-namespace')
+
+# Run workflow
+result = client.execute_workflow(
```

Comment:
Maybe add EOL comments to indicate what arguments are? In particular, the 'Temporal' arg is non-obvious.

Reply:
I don't think it's that common to have to annotate all uses of Ruby code for these arguments. Like Python and Go before it, the argument comes after the workflow but we similarly do not add EOL comments there to say what comes after the workflow is its argument. There will be many many examples and docs showing calling a workflow this way, I am unsure it's reasonable to expect those to all have comments (nor do I think this doc is special, there are going to be lots of quickstart/example docs).